sway-error/src/error.rs

```diff
@@ -3112,6 +3122,32 @@ impl ToDiagnostic for CompileError {
                     help
                 },
             },
+            ABIDuplicateName { span, other_span: other, is_attribute } => Diagnostic {
+                reason: Some(Reason::new(code(1), "Duplicated name found for renamed ABI type.".into())),
+                issue: Issue::error(
+                    source_engine,
+                    span.clone(),
+                    String::new()
+                ),
+                hints: vec![
+                    Hint::help(
+                        source_engine,
+                        other.clone(),
+                        format!("This is the existing {} with conflicting name", if *is_attribute { "attribute" } else { "type" }),
```

Comment:
According to the guidelines, hints shoul end up in punctuation (we have messages that violates that, I know...).
```suggestion
                        format!("This is the existing {} with conflicting name.", if *is_attribute { "attribute" } else { "type" }),
```